spec/operator_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'operator'
+
+describe Operator do
+
+  let(:text) {double(:text)}
+  let(:menu) {double(:menu)}
+  let(:random_message) {'hello'}
+  let(:current_text) {"steak, chips 3, pizza 2"}
+  let(:menu_string) {double(:menu_string)}
+  #let(:message) {"requesting menu please"}
+  subject(:operator) {described_class.new(menu, text)}
+
+  before(:example) do
+    allow(text).to receive(:new_message).and_return(current_text)
+    allow(text).to receive(:send_text)
+    allow(menu).to receive(:food).and_return(menu_string)
+    allow(menu).to receive(:create_receipt).and_return(44)
+  end
+
+  it {is_expected.to respond_to(:send_menu)}
+
+  describe '#check_new_messages' do
+    it 'should be able to check for new incoming text messages' do
+      expect(operator.check_new_messages).to eq current_text
+    end
+  end
+
+  describe '#sort_order' do
+    it 'can sort an incoming order' do
+      # operator.check_new_messages
+      expect(operator.sort_order(current_text)).to eq({:steak=>1,:chips=>3,:pizza=>2})
```

Comment:
Metrics/LineLength: Line is too long. [86/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.<br>Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)